tests/lib/devicetree/devices/testcase.yaml

```diff
@@ -18,3 +18,4 @@ tests:
       - mimxrt595_evk_cm33
       - nrf9131ek_nrf9131
       - nrf9131ek_nrf9131_ns
+      - kincony_kc868_a32
```

Comment:
Why is this added?

Reply:
https://github.com/zephyrproject-rtos/zephyr/actions/runs/7172682929/job/19546571568

Reply:
For some reason, `CONFIG_I2C` is forced to be y when test case sets to n, breaking it. That is weird.

Reply:
Good point, I'll try to check that too.

Reply:
@henrikbrixandersen @sylvioalves I've done a quick investigation and didn't find anything that can be related to it. Can we fix this later as it is a merge blocker?

Reply:
I think we should try to find the root cause of this. 

@nashif Any ideas?

Reply:
@henrikbrixandersen Since nobody has looked at it, can we put it aside?

Reply:
@bbilas Yes. Please open a separate issue for tracking this.